tools/standardize_config_json.py

```diff
@@ -99,29 +94,78 @@ def print_diff(original: collections.OrderedDict, new: collections.OrderedDict):
     o = json.dumps(original, indent=2)
     c = json.dumps(new, indent=2)
 
-    diff = difflib.unified_diff(o.split("\n"), c.split("\n"))
+    diff = list(difflib.unified_diff(o.split("\n"), c.split("\n")))
 
-    LOG.info("\n".join(diff))
+    if len(diff) == 0:
+        LOG.info("No changes")
+    else:
+        LOG.info("Applied diff:")
+        LOG.info("\n".join(diff))
 
 
-def process(filename: str):
-    contents = read_json_file(filename)
-    original = read_json_file(filename)
+def process(service_config_json_path: str, sdk_tools_config: dict):
+    contents = read_json_file(service_config_json_path)
+    original = read_json_file(service_config_json_path)
 
     reordered = reorder_service(contents["properties"]["service"]["properties"])
     contents["properties"]["service"]["properties"] = reordered
 
+    sections = sdk_tools_config.get("sections", {})
+    for section_name, head_and_tail in sections.items():
+        if section_name in contents["properties"]:
+            head = head_and_tail.get("head")
+            tail = head_and_tail.get("tail")
+            properties = contents["properties"][section_name]["properties"]
+            reordered = reorder(properties, head, tail, reorder_property)
+            contents["properties"][section_name]["properties"] = reordered
+
     print_diff(original, contents)
 
-    write_json_file(filename, contents)
+    write_json_file(service_config_json_path, contents)
 
 
-if __name__ == "__main__":
-    files = [
-        "frameworks/cassandra/universe/config.json",
-        "frameworks/hdfs/universe/config.json",
-        "frameworks/helloworld/universe/config.json",
-    ]
+def main(argv):
+    parser = argparse.ArgumentParser(
+        description="Standardizes the ordering of sections in SDK service config.json files."
+    )
+
+    parser.add_argument(
+        "--service-config-json",
+        type=str,
+        required=True,
+        default=None,
+        help="Path to the service configuration JSON file to be standardized",
+    )
+
+    parser.add_argument(
+        "--sdk-tools-config",
+        type=str,
+        required=False,
+        default=None,
+        help="Path to the SDK Tools configuration file",
+    )
 
-    for f in files:
-        process(f)
+    args = parser.parse_args()
+
+    service_config_json_path = args.service_config_json
+    sdk_tools_config_path = args.sdk_tools_config
+
+    if not os.path.isfile(service_config_json_path):
+        LOG.info("'%s' is not a file, was expecting a 'config.json' file", service_config_json_path)
+
+    if sdk_tools_config_path:
+        if not os.path.isfile(sdk_tools_config_path):
+            LOG.info(
+                "'%s' is not a file, was expecting a 'sdk-tools.json' file", sdk_tools_config_path
+            )
+
+        LOG.info("Parsing SDK tooling configuration from '%s'", sdk_tools_config_path)
+        sdk_tools_config = json.loads(read_file(sdk_tools_config_path))
```

Comment:
.. then this is going to crash right here, no?

Reply:
Of course! deb7333e0f92c50103625f85493b0f5a5d4cc48f